### ISSUE TITLE
Server falls asleep with active SSH connection

### ISSUE THREAD

**Issue by eamodio** [PowerShell/Win32-OpenSSH#1712]
Refs: https://github.com/microsoft/vscode-remote-release/issues/4097

**"OpenSSH for Windows" version**
Server: `7.7.2.3`, Client: `8.1.0.1`
 
**Server OperatingSystem**
Windows 10.0.19042 Pro (Beta Channel)

**Client OperatingSystem**
Windows 10.0.20270 Enterprise (Dev Channel)

**What is failing**
Server falls asleep during active SSH connection -- happens with both an active terminal connection and using VS Code remote

**Expected output**
Server should stay awake

**Comment by mgkuhn** [comment]
By “Server falls asleep”, do you mean its operating system enters a [system power state](https://docs.microsoft.com/en-us/windows/win32/power/system-power-states) change (sleep (S1-S3), hibernate (S4))? How have you verified this? Does the server still [ping](https://kb.iu.edu/d/aopy) (reply to ICMP echo request), if it did before? See also [System Sleep Criteria](https://docs.microsoft.com/en-us/windows/win32/power/system-sleep-criteria).

Or does your network connection go away and the server didn't actually sleep? Many network-address translation boxes and similar forget the translation after a couple of minutes or hours, and long-lived ssh connections are often the victim.

What are the  `TCPKeepAlive` options in the ssh client and sshd server, and does changing them have any effect?

**Comment by eamodio** [comment]
Yeah, S1-S3 (not hibernate). I verify it by hearing the fan on my desktop machine turn off and if I wake it via the keyboard/mouse, or by sending a magic packet, everything will work again.

Where do I find the `TCPKeepAlive` setting?

**Comment by mgkuhn** [comment]
Type [man ssh_config](https://linux.die.net/man/5/ssh_config) and [man sshd_config](https://linux.die.net/man/5/sshd_config) in [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10). (Windows does not come with Unix-style built-in man pages, so best read them on Linux.)

**Comment by davidgarciaMontreal** [comment]
Here is  an example that applies to all Hosts.
  [using ssh config file](https://linuxize.com/post/using-the-ssh-config-file/)
```bash
# file: ~/.ssh/config
# chmod 600 ~/.ssh/config
# # chown $USER ~/.ssh/config
Host *
    ServerAliveInterval 240
```

**Comment by eamodio** [comment]
@davidgarciaMontreal Thanks!

@mgkuhn After setting `ServerAliveInterval` I didn't notice any change -- my desktop machine still fell asleep with an ssh connection.

**Comment by mgkuhn** [comment]
What *power plan* are you currently using?

You can query or change your operating system's power plan/scheme via either the Control Panel / Power Options GUI, or on the command line via [`powercfg`](https://docs.microsoft.com/en-us/windows-hardware/design/device-experiences/powercfg-command-line-options) (or the corresponding GUI settings).

I somewhat doubt this is really an OpenSSH issue, as opposed to one of the power-savings configuration of your operating system.

**Comment by JackCA** [comment]
I'm also running into this issue.

I have multiple ssh connections to the windows 10 host, have setup `TCPKeepAlive`, `ServerAliveInterval`, `ClientAliveInterval`, and the machine still sleeps.

My power mode is "high performance".

Here is the output from `powercfg -requests` while there are active connections:
```
PS C:\Users\j> powercfg -requests
DISPLAY:
None.

SYSTEM:
None.

AWAYMODE:
None.

EXECUTION:
None.

PERFBOOST:
None.

ACTIVELOCKSCREEN:
None.
```
